File: libReQLTests/Connection.m
//
//  Connection.m
//  libReQL
//
//  Created by Adam Grandquist on 5/30/15.
//  Copyright (c) 2015 Adam Grandquist. All rights reserved.
//

#import <Cocoa/Cocoa.h>
#import <XCTest/XCTest.h>
#import <libReQL.h>

@interface Connection : XCTestCase

@end

@implementation Connection

- (void)testExample {
  ReQLConnection *connection = [ReQLConnection new];
  XCTAssert([connection isOpen]);
  if ([connection isOpen]) {
    ReQLCursor *cursor = [[ReQLQuery add:@[@2, @2]] run:connection];
    XCTAssert(cursor);
    if (cursor) {
      NSArray *res = [cursor toArray];
      NSError *err;
      XCTAssert(![cursor error:&err]);
      XCTAssert([res isEqualTo:@[@4]]);
    }
    [cursor close];
  }
  [connection close];
}

@end

File: libReQLTests/Connection.swift
// Copyright 2015 Adam Grandquist

import Cocoa
import Quick
import Nimble
import libReQL

class Connection : QuickSpec {
  override func spec() {
    describe("example") {
      it("of connection") {
        let connection = ReQLConnection()
        expect(connection.isOpen()) == true
        if (connection.isOpen()) {
          let cursor = ReQLQuery.newWithNumber(2)!.add([2])!.run(connection)
          expect(cursor!.isOpen()) == true
          if (cursor!.isOpen()) {
            let res = cursor!.toArray()
            expect(res).to(contain(4))
            expect(res).to(haveCount(1))
          }
          cursor!.close()
        }
        connection.close()
      }
    }
  }
}